packages/connect-webextension/src/contentScript.ts

```diff
@@ -1,81 +1,103 @@
 import { WindowServiceWorkerChannel } from '@trezor/connect-web/src/channels/window-serviceworker';
 import { POPUP } from '@trezor/connect/src/events/popup';
+import { CONTENT_SCRIPT_VERSION } from '@trezor/connect/src/data/version';
 
-/**
- * communication between service worker and both webextension and popup manager
- */
-const channel = new WindowServiceWorkerChannel({
-    name: 'trezor-connect',
-    channel: {
-        here: '@trezor/connect-content-script',
-        peer: '@trezor/connect-webextension',
-    },
-});
-
-/**
- * messages that were sent before the channel was initialized
- */
-const messagesQueue: any[] = [];
-let channelReady = false;
-
-/**
- * Firefox enforces some restrictions on the content script that force us to use clones of objects when passing them between the content script and the background script
- */
-function clone(obj: any) {
-    return JSON.parse(JSON.stringify(obj));
-}
-
-/*
- * Passing messages from popup to service worker
- */
-window.addEventListener('message', event => {
-    if (
-        event.data?.channel?.here === '@trezor/connect-webextension' ||
-        event.data?.type === POPUP.CONTENT_SCRIPT_LOADED
-    ) {
+function trezorContentScript() {
+    // Check if extension ID matches the popup URL
+    const urlParams = new URLSearchParams(window.location.search);
+    const targetExtensionId = urlParams.get('extension-id');
+    if (targetExtensionId && targetExtensionId !== chrome.runtime.id) {
         return;
     }
-    if (event.data?.type === POPUP.LOADED) {
-        window.postMessage(
-            {
-                type: POPUP.CONTENT_SCRIPT_LOADED,
-                payload: { ...chrome.runtime.getManifest(), id: chrome.runtime.id },
-            },
-            window.location.origin,
+    const targetContentScriptVersion = urlParams.get('cs-ver');
+    if (targetContentScriptVersion !== CONTENT_SCRIPT_VERSION.toString()) {
+        console.error(
+            'Content script version mismatch! Make sure to update content script to the latest version from @trezor/connect-webextension.',
```

Comment:
hmm but here it still would appear inside popup dev console and I doubt that 3rd party devs check it. Doesn't it make sense to put it to their main area of focus (where npm package is installed)? doesn't this content-script-loading event reach it as well?

Reply:
Currently no, but we can make it happen